src/browser/wisp_network.js

```diff
@@ -0,0 +1,571 @@
+
+
+            break;
+        case "DATA":
+
+            fullPacket = new Uint8Array(5 + frameObj.data.length);
+            view = new DataView(fullPacket.buffer);
+            view.setUint8(0, 0x02);                     // TYPE
+            view.setUint32(1, frameObj.streamID, true); // Stream ID
+            fullPacket.set(frameObj.data, 5);           // Actual data
+
+            break;
+        case "CLOSE":
+            fullPacket = new Uint8Array(5 + 1);
+            view = new DataView(fullPacket.buffer);
+            view.setUint8(0, 0x04);                     // TYPE
+            view.setUint32(1, frameObj.streamID, true); // Stream ID
+            view.setUint8(5, frameObj.reason);          // Packet size
+
+            break;
+
+    }
+    sendPacket(fullPacket, frameObj.type, frameObj.streamID);
+}
+
+/**
+ * @constructor
+ *
+ * @param {BusConnector} bus
+ * @param {*=} config
+ */
+function WispNetworkAdapter(wispURL, bus, config)
+{
+    const registerWS = function() {
+        wispws = new WebSocket(wispURL.replace("wisp://", "ws://").replace("wisps://", "wss://"));
+        wispws.binaryType = "arraybuffer";
+        wispws.onmessage = (event) => {
+            processIncomingWispFrame(new Uint8Array(event.data));
+        };
+        wispws.onerror = () => {
+            registerWS();
+        };
+        wispws.onclose = () => {
+            registerWS();
+        };
+    };
+    registerWS();
+    config = config || {};
+    this.bus = bus;
+    this.id = config.id || 0;
+    this.router_mac = new Uint8Array((config.router_mac || "52:54:0:1:2:3").split(":").map(function(x) { return parseInt(x, 16); }));
+    this.router_ip = new Uint8Array((config.router_ip || "192.168.86.1").split(".").map(function(x) { return parseInt(x, 10); }));
+    this.vm_ip = new Uint8Array((config.vm_ip || "192.168.86.100").split(".").map(function(x) { return parseInt(x, 10); }));
+    this.masquerade = config.masquerade === undefined || !!config.masquerade;
+    this.vm_mac = new Uint8Array(6);
+
+    this.tcp_conn = {};
+
+    this.bus.register("net" + this.id + "-mac", function(mac) {
+        this.vm_mac = new Uint8Array(mac.split(":").map(function(x) { return parseInt(x, 16); }));
+    }, this);
+    this.bus.register("net" + this.id + "-send", function(data)
+    {
+        this.send(data);
+    }, this);
+}
+
+WispNetworkAdapter.prototype.destroy = function()
+{
+};
+
+// https://stackoverflow.com/questions/4460586/javascript-regular-expression-to-check-for-ip-addresses
+function validateIPaddress(ipaddress) {
+    if(/^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/.test(ipaddress)) {
+      return true;
+    }
+    return false;
+}
+
+// DNS over HTTPS fetch, recursively fetch the A record until the first result is an IPv4
+async function dohdns(q) {
+    const prefered_fetch = (window.anura?.net?.fetch) || fetch;
```

Comment:
What is this about?

Reply:
That was an accident I left from a seperate project, will fix